### PR TITLE
feat: Add internal debug endpoint for memory retrieval

### LINKED ISSUE #43: Add internal debug endpoint for memory retrieval
## Context
Memory accuracy is hard to tune without visibility. Product decision: raw memory content may be logged and shown in internal debug output for now, with the expectation this can be restricted later.

## Scope
Add an internal HTTP endpoint to inspect what memories would be retrieved for a given user, companion, and message. This endpoint is for development/debugging only.

Suggested endpoint:
`POST /debug/memory/retrieve`

Suggested request:
```json
{
  "user_mail_id": "john@example.com",
  "ai_companion_id": 1,
  "user_message": "I had a rough day and need you to remember what I like."
}
```

Suggested response:
```json
{
  "user_mail_id": "john@example.com",
  "ai_companion_id": 1,
  "memories": [
    {"memory_id": 12, "memory_type": "preference", "score": 0.91, "content": "..."}
  ]
}
```

Use existing user/companion ownership validation from websocket flow. Unknown user or invalid companion should return standard errors.

## Acceptance Criteria
- Endpoint validates user exists and owns the companion.
- Endpoint returns retrieved memory IDs, types, scores, canonical keys, source, and raw content.
- Endpoint never returns another companion's memories.
- Endpoint is disabled or guarded by config in production-like settings if `MISTRIA_MEMORY_DEBUG_ENDPOINT_ENABLED=false`.
- Tests cover valid request, invalid user, invalid companion ownership, and no-memory result.

### PR DESCRIPTION
**Pull request by umang-910**
Resolves #43. Adds POST /debug/memory/retrieve endpoint guarded by MISTRIA_MEMORY_DEBUG_ENDPOINT_ENABLED. Includes full ownership validation and tests.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -282,6 +283,37 @@ def get_ai_companion(ai_companion_id: int) -> AICompanionResponse:
...
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="Endpoint disabled or memory system not configured"
+        )
+
+    user = user_repository.find_by_email(payload.user_mail_id)
+    if not user:
+        raise CompanionNotFoundError("User not found.")
+
+    companion = ai_companion_repository.get_latest_ai_companion(user.id, payload.ai_companion_id)
```
Comment (vatsal1306) [inline main.py]: `SQLiteAICompanionRepository` does not have method `get_latest_ai_companion()`. So this endpoint will raise `AttributeError`. The existing websocket flow uses
```python
ai_companion = self.ai_companion_repo.find_by_id(request.ai_companion_id)
if not ai_companion or ai_companion.user_id != user.id:
    ...
```

File: main.py [hunk 2]
```diff
@@ -282,6 +283,37 @@ def get_ai_companion(ai_companion_id: int) -> AICompanionResponse:
...
+    if not companion:
+        raise CompanionNotFoundError(f"Companion {payload.ai_companion_id} not found or not owned by user.")
+
+    memories = await memory_service.retrieve_memories(
+        user_id=user.id,
+        ai_companion_id=companion.id,
+        query_text=payload.user_message,
+        limit=settings.memory.retrieval_top_k,
+    )
```
Comment (vatsal1306) [inline main.py:307]: `retrieve_memories()` does not have any params named `query_text` or `limit`. So the endpoint will fail at runtime with an unexpected keyword argument error.

File: tests/unit/backend/test_main_endpoints.py [hunk 3]
```diff
@@ -222,3 +223,97 @@ async def test_companion_endpoints_delegate_to_service(monkeypatch):
...
+    mock_settings = mock.Mock()
+    mock_settings.memory.debug_endpoint_enabled = True
+    monkeypatch.setattr(main, "settings", mock_settings)
+    monkeypatch.setattr(main, "memory_service", mock.Mock())
+
+    user_repo = mock.Mock()
+    user_repo.find_by_email.return_value = sample_user
+    monkeypatch.setattr(main, "user_repository", user_repo)
+
+    companion_repo = mock.Mock()
```
Comment (vatsal1306) [inline tests/unit/backend/test_main_endpoints.py]: The tests use loose `mock.Mock()` objects for `ai_companion_repository` and `memory_service`, then assert calls against methods/kwargs that do not exist in code repo. 
```python
companion_repo.get_latest_ai_companion.return_value = mock_companion
```
```python
memory_service.retrieve_memories.assert_awaited_once_with(
    user_id=sample_user.id,
    ai_companion_id=mock_companion.id,
    query_text="hello",
    limit=5,
)
```
This is why the test passes while the actual endpoint failed. 
Suggested fix: Use mocks with spec/autospec from the real repository/service interfaces, or test the endpoint with the actual method names. Also add the missing acceptance test for a valid request that returns an empty `memories: []` result

**Review by vatsal1306 — CHANGES_REQUESTED** [review]

**Review by vatsal1306 — APPROVED** [review]
